spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala

```diff
@@ -146,140 +150,55 @@ class RNN(
     }
   }
 
-  private def initMemoryDescs(inputs: Array[MemoryData]) = {
-    // TODO: The default format of input is TNC
+  override private[mkldnn] def initFwdPrimitives(inputs: Array[MemoryData], phase: Phase) = {
+    val kind = if (!isTraining()) {
+      PropKind.ForwardInference
+    } else {
+      PropKind.ForwardTraining
+    }
+
     /**
-     * The default format of input is TNC.
+     * TODO: The default format of input is TNC
      * Batch size of input is needed by creating memory descriptors of src iter and dst iter.
      * Step size of input is needed by creating memory descriptor of dst layer.
      * By default, batch size of input is the second element of inputShape
      * and step size is the first element of inputShape.
      */
-    val(inputShape, inputLayout) = inputs(0).layout match {
-      case Memory.Format.tnc => /* tnc */
-        (inputs(0).shape, Memory.Format.tnc)
+
+    val inputShape = inputs(0).layout match {
```

Comment:
Please also support ntc

Reply:
Included case for ntc. Batch size and step size are assigned values according to the input format. The input shape in RNN is still tnc, is there any problem?